### PR TITLE
Handle routers which do not allow runtime configuration

### PR DESCRIPTION
**Pull request by DASPRiD**
Related PRs:

https://github.com/zendframework/zend-expressive-router/pull/6
https://github.com/zendframework/zend-expressive-helpers/pull/9

### CODE REVIEW

File: test/ApplicationTest.php [hunk 1]
```diff
@@ -149,6 +150,14 @@ public function testCallingRouteWithOnlyAPathRaisesAnException()
         $app->route('/path');
     }
 
+    public function testCallingRouteWithConsumerOnlyRouter()
+    {
+        $this->router = $this->prophesize(RouteInterface::class);
```
Comment (samsonasik) [inline test/ApplicationTest.php]: s/`RouteInterface::class`/`RouterInterface::class`

Comment (DASPRiD) [inline test/ApplicationTest.php]: Sneaky little errors, hehe, thanks.

**Comment by michaelmoussa** [comment]
@DASPRiD can you explain this PR a little more? I'm not clear on why it's needed (or if it is even needed anymore). If we take the approach I mention in [this comment](https://github.com/zendframework/zend-expressive-router/pull/6#issuecomment-240421488), then any `RouterInterface`'s `generateUri(...)` method will accept the optional `$options` param, and there'd be no need to distinguish between a plain `RouterInterface` and a `RuntimeConfigurableRouterInterface`.

**Comment by michaelmoussa** [comment]
Closing w/o merge as this will not be necessary due to [zend-expressive-router#6](https://github.com/zendframework/zend-expressive-router/pull/6) - all `RouterInterface` instances will still have the `addRoute` method.

**Pull request closed without merging**